src/cloudflare/internal/vectorize-api.ts

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+/// <reference path="./vectorize.d.ts" />
+
+interface Fetcher {
+  fetch: typeof fetch;
+}
+
+enum Operation {
+  INDEX_GET,
+  VECTOR_QUERY,
+  VECTOR_INSERT,
+  VECTOR_UPSERT,
+  VECTOR_GET,
+  VECTOR_DELETE,
+}
+
+class VectorSearch {
+  public static async send(
+    fetcher: Fetcher,
+    operation: Operation,
+    endpoint: string,
+    init: RequestInit
+  ): Promise<Response> {
+    const res = await fetcher.fetch(
+      `http://vector-search/binding/${endpoint}`,
```

Comment:
A comment here indicating that `http://vector-search/...` is a dummy prefix here would be helpful, along with perhaps a brief explanation how that is used.